moviebot/controller/http_data_formatter.py

```diff
@@ -0,0 +1,133 @@
+
+from dataclasses import asdict, dataclass, field
+from typing import Any, Dict, List
+
+from moviebot.agent.agent import DialogueOptions
+
+HTTP_OBJECT_MESSAGE = Dict[str, Dict[str, str]]
+
+SHORT_ANSWER = {
+    "I like this recommendation.": "I like this",
+    "I have already watched it.": "Seen it",
+    "Tell me more about it.": "Tell me more",
+    "Recommend me something else please.": "Something else",
+    "Tell me something about it.": "More information",
+    "/restart": "Restart",
+    "I would like a similar recommendation.": "Similar",
+    "I want to restart for a new movie.": "Restart",
+    "I would like to quit now.": "Quit",
+}
+
+
+def shorten(input: str) -> str:
+    """Creates shorter versions of agent responses.
+
+    Args:
+        input: Agent response.
+
+    Returns:
+        Shorter version of input message.
+    """
+    return SHORT_ANSWER.get(input, input)
+
+
+@dataclass
+class Message:
+    text: str
+    intent: str
+    attachment: Dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass
+class Button:
+    title: str
+    payload: str
+    button_type: str = field(default="postback")
+
+
+def create_buttons(user_options: DialogueOptions) -> List[Dict[str, Any]]:
+    """Creates a list of buttons for each agent's option.
+
+    Args:
+        user_options: Agent's options
+
+    Returns:
+        List of buttons objects.
+    """
+    options = []
+    for option in user_options.values():
+        if isinstance(option, str):
+            option = [option]
+        for item in option:
+            button = Button(
+                title=shorten(item),
+                payload=item,
+                button_type="postback",
+            )
+            options.append(asdict(button))
+    return options
+
+
+def text_message(
+    user_id: str, message: str, intent: str = "UNK"
+) -> HTTP_OBJECT_MESSAGE:
+    """Creates a message with a text response.
+
+    Args:
+        user_id: Id of the recipient.
+        message: Message to send.
+        intent: Intent of the message. Defaults to 'UNK'.
+
+    Returns:
+        Object to send to Flask server.
+    """
+    message = Message(text=message, intent=intent)
+    text = {
+        "recipient": {"id": user_id},
+        "message": asdict(message),
+    }
+    return text
+
+
+def buttons_message(
+    user_id: str, buttons: List[Dict[str, Any]], text: str, intent="UNK"
+) -> HTTP_OBJECT_MESSAGE:
+    """Creates a message along with a list of buttons.
+
+    Args:
+        user_id: Id of the recipient.
+        buttons: List of buttons.
+        text: Message to send.
+        intent: Intent of the message.
+
+    Returns:
+        Object with message and buttons to send to Flask server.
+    """
+    attachment = {
+        "type": "buttons",
+        "payload": {
+            "buttons": buttons,
+        },
+    }
+    message = Message(text=text, intent=intent, attachment=attachment)
+    template = {
```

Comment:
The template and message lines are very similar to the ones in `text_message` and maybe updated `movie_message`. Can it be factored out?